parsel/selector.py

```diff
@@ -139,9 +139,9 @@ class Selector(object):
     selectorlist_cls = SelectorList
 
     def __init__(self, text=None, type=None, namespaces=None, root=None,
-                 base_url=None, _expr=None):
+                 base_url=None, _expr=None, parser_cls=None):
```

Comment:
I tried using `lxml.html.html5parser.HTMLParser` but unfortunately, `encoding` param makes it choke
```
>>> from lxml.html.html5parser import HTMLParser
>>> s = Selector(text=u'<html><body><p>test</p></body></html>', parser_cls=HTMLParser)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/paul/src/parsel/parsel/selector.py", line 151, in __init__
    root = self._get_root(text, base_url)
  File "/home/paul/src/parsel/parsel/selector.py", line 162, in _get_root
    return create_root_node(text, self._parser, base_url=base_url)
  File "/home/paul/src/parsel/parsel/selector.py", line 42, in create_root_node
    parser = parser_cls(recover=True, encoding='utf8')
  File "/home/paul/.virtualenvs/parseldev/lib/python3.4/site-packages/lxml/html/html5parser.py", line 32, in __init__
    _HTMLParser.__init__(self, strict=strict, tree=TreeBuilder, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```